### PR TITLE
[CodeStyle][CINN] fix cinn cpplint codestyle 50-55

### PR DESCRIPTION
**Pull request by gouzil**
### PR types

Others
### PR changes

Others
### Description

* fix cinn cpplint codestyle errors about [whitespace/empty_if_body], [whitespace/comments], [whitespace/blank_line], [runtime/init], [runtime/casting], [build/storage_class]
* https://github.com/PaddlePaddle/Paddle/issues/54953

### CODE REVIEW

**Review by SigureMo — DISMISSED** [review]
LGTM

File: paddle/utils/variant.h [hunk 1]
```diff
@@ -803,7 +803,6 @@ inline constexpr T *addressof(T &arg) noexcept {
 
 template <typename T>
 inline constexpr T *addressof(const T &&) = delete;
-
```
Comment (SigureMo) [inline paddle/utils/variant.h]: image

非 CINN 的不用改

Comment (gouzil) [reply paddle/utils/variant.h]: 回滚了

**Review by SigureMo — APPROVED** [review]
LGTM，麻烦 rerun 下 PR-CI-ROCM-Compile

**Comment by gouzil** [comment]
> LGTM，麻烦 rerun 下 PR-CI-ROCM-Compile

Done

**Pull request merged**